<launch>
  <!-- HoloLens depth data receiver -->
  <include file="$(find hololens_depth_data_receiver)/launch/depth_data_receiver.launch"/>
  <node name="hololens_rviz_tf1" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 1.5707963 /odom /hololens_world 100"/>
  <node name="hololens_rviz_tf2" pkg="tf" type="static_transform_publisher" args="0 0 0 -1.5707963 0 0 /hololens /odom 100"/>

  <!-- FLOBOT 3D Object Detector -->
  <node pkg="object3d_detector" type="object3d_detector" name="object3d_detector" output="screen" respawn="false">
    <param name="model_file_name" type="string" value="$(find object3d_detector)/libsvm/pedestrian_hololens.model"/>
    <param name="range_file_name" type="string" value="$(find object3d_detector)/libsvm/pedestrian_hololens.range"/>

    <param name="print_fps" type="bool" value="false"/>
    <param name="frame_id" type="string" value="odom"/>

    <param name="cluster_size_min" type="int" value="5"/>
    <param name="cluster_size_max" type="int" value="30000"/>
    <param name="human_probability" type="double" value="0.7"/>
  </node>
  
  <!-- NBellotto's Bayes People Tracker -->
  <rosparam command="load" file="$(find object3d_detector)/config/object3d_detector.yaml"/>
  <node pkg="bayes_people_tracker" type="bayes_people_tracker" name="bayes_people_tracker" output="screen" respawn="false">
    <param name="base_link" type="string" value="odom"/>
    <param name="target_frame" type="string" value="odom"/>
  </node>
  
  <!-- ROS Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find object3d_detector)/launch/object3d_detector.rviz"/>
</launch>
